App/ViewModels/AppShellViewModel.cs

```diff
@@ -310,6 +318,13 @@ private void OnNotificationOpened(object sender, FirebasePushNotificationRespons
                 await UpdateDeals();
             });
 
+            // Update notification count
+            int count = (Preferences.Get(AppConstant.NotificationCount, 1)) - 1;
+            if (count < 0)
+                count = 0;
+            Preferences.Set(AppConstant.NotificationCount, count);
+            Badge.Default.SetCount((uint)count);
```

Comment:
I don't mind repeating myself here.
I wouldn't see the point of creating a method just for this at the moment.

maybe later